### PR TITLE
Fix exception wrong place and entity schematics serialization

### PR DESCRIPTION
**Pull request by afdw**

### CODE REVIEW

**Review by AlexIIL — CHANGES_REQUESTED** [review]
Looks good! Minor issue though: all readFromNbt methods should throw ````InvalidDataInputException```` rather than just ````Exception````

File: api/buildcraft/api/schematics/ISchematicEntity.java [hunk 1]
```diff
@@ -29,4 +28,9 @@
     Entity build(World world, BlockPos basePos);
 
     Entity buildWithoutChecks(World world, BlockPos basePos);
+
+    NBTTagCompound serializeNBT();
+
+    /** @throws Exception If the input data wasn't correct or didn't make sense. */
+    void deserializeNBT(NBTTagCompound nbt) throws Exception;
```
Comment (AlexIIL) [inline api/buildcraft/api/schematics/ISchematicEntity.java]: Specifically here

File: api/buildcraft/api/schematics/ISchematicBlock.java [hunk 2]
```diff
@@ -1,18 +1,16 @@
...
-
 import net.minecraftforge.fluids.FluidStack;
 
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Set;
+
 public interface ISchematicBlock<S extends ISchematicBlock<S>> {
     int BLOCK_LEVEL = 0;
     int FLUID_LEVEL = 1;
```
Comment (AlexIIL) [inline api/buildcraft/api/schematics/ISchematicBlock.java:17]: Specifically below this point (````deserialiseNBT````)

**Review by AlexIIL — CHANGES_REQUESTED** [review]
Throw InvalidDataInputException rather than any Exception

**Comment by afdw** [comment]
@AlexIIL done

**Review by AlexIIL — APPROVED** [review]

**Comment by AlexIIL** [comment]
Thanks

**Pull request merged**